site/app/views/submission/HomeworkView.php

```diff
@@ -1214,9 +1214,140 @@ function downloadFile(file, path) {
             }
             $return .= <<<HTML
 </div>
+  <div class="content"> 
+HTML;
+    $return .= $this->core->getOutput()->renderTemplate('submission\Homework', 'showRequestForm', $gradeable);
+    if($gradeable->getRegradeStatus() !== 0)
+      $return .= $this->core->getOutput()->renderTemplate('submission\Homework', 'showRegradeDiscussion', $gradeable);
+    $return .= <<<HTML
+  </div>
 HTML;
         }
 
         return $return;
     }
+    public function showRequestForm($gradeable){
+      $return = <<<HTML
+      <div class = "sub">
+        <div style="float: left; width: 50%"><h3>Regrade Discussion</h3></div>
+HTML;
+        $is_disabled = "";
+        $action = "";
+        $url = "";
+        $class = "btn-default";
+        if($gradeable->getRegradeStatus() === 0){
+          $message = "Request Regrade";
+          $action = "showPopUp()";
```

Comment:
I am getting an Uncaught ReferenceError when I try to click the Request Regrade button as below:
![regraderequestuncaughtreferenceerror](https://user-images.githubusercontent.com/11095297/40629833-c174b0c2-629b-11e8-9aed-6563175ba953.JPG)
It looks like the cause is that the ShowPopUp method is never defined. Maybe it was lost in the merging.

Reply:
Whoops looks like I accidentally deleted that bit of code when I was cleaning it up. Should be good now. 
By design, the regrading interface was supposed to be hidden unless a student requests a regrade, once there is a request in process the yellow toolbar will have an additional button (right now its just a "!") which will open up another box for handling regrade requests. 
The design still needs a bit of finalization on the grader's side.